### PR TITLE
boards/sim: Defconfigs for nand and mnemofs

### PR DESCRIPTION
**Pull request by resyfer**
## Summary
Adds nand and mnemofs sim configs.

## Impact
Testers and users have default configuration files to use nand device simulators out of the box.

## Testing
Manual testing using make after applying defconfigs.

### CODE REVIEW

**Comment by acassis** [comment]
@resyfer please fix the compilation error:
```
Configuration/Tool: sim/nand
2024-08-28 20:23:32
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Building NuttX...
Error: mtd/mtd_nand.c:82:16: error: 'nand_devscan' declared 'static' but never defined [-Werror=unused-function]
   82 | static int     nand_devscan(FAR struct nand_dev_s *nand);
      |                ^~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:102: mtd_nand.o] Error 1
mtd/mtd_nandram.c: In function 'nand_ram_storage_status':
Error: mtd/mtd_nandram.c:184:13: error: variable 'bad' set but not used [-Werror=unused-but-set-variable]
  184 |   uint8_t   bad;
      |             ^~~
Error: mtd/mtd_nandram.c:183:13: error: variable 'erases' set but not used [-Werror=unused-but-set-variable]
  183 |   uint16_t  erases;
      |             ^~~~~~
Error: mtd/mtd_nandram.c:182:13: error: variable 'writes' set but not used [-Werror=unused-but-set-variable]
  182 |   uint16_t  writes;
      |             ^~~~~~
Error: mtd/mtd_nandram.c:181:13: error: variable 'reads' set but not used [-Werror=unused-but-set-variable]
  181 |   uint16_t  reads;
      |             ^~~~~
mtd/mtd_nandram.c: At top level:
Error: mtd/mtd_nandram.c:178:13: error: 'nand_ram_storage_status' defined but not used [-Werror=unused-function]
  178 | static void nand_ram_storage_status(void)
      |             ^~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Review by acassis — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@resyfer please rebase to the last master which fix the ci issue.

**Comment by resyfer** [comment]
> @resyfer please rebase to the last master which fix the ci issue.

@xiaoxiang781216 Alright thanks a lot. Pushed after rebase.

File: fs/mnemofs/mnemofs.c [hunk 1]
```diff
@@ -579,7 +579,7 @@ static ssize_t mnemofs_read(FAR struct file *filep, FAR char *buffer,
                           buffer, buflen);
   if (ret < 0)
     {
-      finfo("Error while reading. Ret: %d.", ret);
+      finfo("Error while reading. Ret: %ld.", ret);
```
Comment (xiaoxiang781216) [inline fs/mnemofs/mnemofs.c:582]:
```suggestion
      finfo("Error while reading. Ret: %zd.", ret);
```

File: fs/mnemofs/mnemofs.c [hunk 2]
```diff
@@ -593,7 +593,7 @@ static ssize_t mnemofs_read(FAR struct file *filep, FAR char *buffer,
   nxmutex_unlock(&MFS_LOCK(sb));
 
 errout:
-  finfo("Mnemofs read exited with %d.", ret);
+  finfo("Mnemofs read exited with %ld.", ret);
```
Comment (xiaoxiang781216) [inline fs/mnemofs/mnemofs.c:596]: please change ALL size_t/ssize_ to zu/zd in your code.

**Comment by resyfer** [comment]
@xiaoxiang781216 Alright thanks for the suggestion! I've updated the same.

File: fs/mnemofs/mnemofs_journal.c [hunk 3]
```diff
@@ -655,7 +654,8 @@ int mfs_jrnl_wrlog(FAR struct mfs_sb_s * const sb,
 
   /* Store */
 
-  jrnl_blk = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).log_cblkidx);
+  /* jrnl_blk = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).log_cblkidx); */
```
Comment (xiaoxiang781216) [inline fs/mnemofs/mnemofs_journal.c:657]: why not remove the unused statement directly

File: fs/mnemofs/mnemofs_master.c [hunk 4]
```diff
@@ -206,7 +205,9 @@ int mfs_mn_init(FAR struct mfs_sb_s * const sb, const mfs_t jrnl_blk)
   struct mfs_jrnl_log_s log;
 
   mblk1       = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).n_blks);
-  mblk2       = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).n_blks + 1);
+
+  /* mblk2       = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).n_blks + 1); */
```
Comment (xiaoxiang781216) [inline fs/mnemofs/mnemofs_master.c:209]: ditto

File: fs/mnemofs/mnemofs_master.c [hunk 5]
```diff
@@ -373,9 +372,11 @@ int mfs_mn_move(FAR struct mfs_sb_s * const sb, struct mfs_ctz_s root,
       /* TODO: Move journal. Master blocks are full. */
     }
 
-  mblk1 = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).n_blks);
-  mblk2 = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).n_blks + 1);
-  mn    = MFS_MN(sb);
+  /* mblk1 = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).n_blks);
+   * mblk2 = mfs_jrnl_blkidx2blk(sb, MFS_JRNL(sb).n_blks + 1);
```
Comment (xiaoxiang781216) [inline fs/mnemofs/mnemofs_master.c:376]: ditto

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by acassis** [comment]
@xiaoxiang781216 I think the CI was failing because %llx on MacOS, I think now it will break all new PRs

**Comment by acassis** [comment]
> @xiaoxiang781216 I think the CI was failing because %llx on MacOS, I think now it will break all new PRs

Nevermid, Saurav said the issue was fixed

**Pull request merged**